# Yug Dedhia
# 2020115004
## BRED ASSIGNMENT 2 Part 2

In [1]:
import pandas as pd
import pingouin as pg
import seaborn as sns
import matplotlib.pyplot as plt
import numpy as np

/home/yugdedhia/.local/lib/python3.10/site-packages/outdated/utils.py:14: OutdatedPackageWarning: The package outdated is out of date. Your version is 0.2.1, the latest is 0.2.2.
Set the environment variable OUTDATED_IGNORE=1 to disable these warnings.
  return warn(


In [2]:
xls = pd.ExcelFile('Likert_Scale_Ratings.xlsx')
df1 = pd.read_excel(xls, 'Pleasantness')
df2 = pd.read_excel(xls, 'Energy')
#df = pd.read_excel('Likert_Scale_Ratings.xlsx')
df2

Energy Ratings  Unnamed: 1  Unnamed: 2  Unnamed: 3  Unnamed: 4  \
0                 1           2           4           1           2   
1                 1           3           3          -1           1   
2                 1          -2          -1          -1           1   
3                 1          -2           2          -1           1   
4                 1           2          -1          -1          -1   
..              ...         ...         ...         ...         ...   
105               2           4           4          -2          -2   
106               2           1           4           1           1   
107               1          -3           3           2           0   
108               2           2           4           1           0   
109               2           4           4           1           0   

     Unnamed: 5  Unnamed: 6  Unnamed: 7  Unnamed: 8  Unnamed: 9  ...  \
0             1          -1           2           1          -2  ...   
1             2           4          -1           3           2  ...   
2            -2          -1          -3          -1           1  ...   
3            -2          -1           2           3           2  ...   
4            -3           2          -1          -2          -2  ...   
..          ...         ...         ...         ...         ...  ...   
105           4           4           3           4           3  ...   
106          -1          -1           4          -3          -2  ...   
107          -2           1           2           1          -1  ...   
108          -1           1           2           4           2  ...   
109           2           4           3           4           2  ...   

     Unnamed: 20  Unnamed: 21  Unnamed: 22  Unnamed: 23  Unnamed: 24  \
0              2            2            2           -2            2   
1              1            2            3            3            1   
2             -1            1           -1            2           -1   
3              1            0            2            2            1   
4             -1            0            1            1            2   
..           ...          ...          ...          ...          ...   
105            2            2            2            2            3   
106            1            1            1            1            2   
107            1            1            2            1            1   
108            2            2            3            2            3   
109            2            3            3            3            3   

     Unnamed: 25  Unnamed: 26  Unnamed: 27  Unnamed: 28  Unnamed: 29  
0              3            1            1            3            4  
1              3           -2            4            1            2  
2              3            0           -2           -1           -2  
3              3            1            1            2            3  
4              1            0           -1            1           -3  
..           ...          ...          ...          ...          ...  
105            3            3            3            3            4  
106            3            0            3            1            2  
107            3            2            2            2            3  
108            2            0            3            3            3  
109            4            3            4            2            4  

[110 rows x 30 columns]

In [34]:
##exchanging rows and columns
df1_t = df.T
df1_t

0    1    2    3    4    5    6    7    8    9    ...  \
Pleasantness Ratings   -1   -1   -1   -1   -1   -3   -1   -2   -1   -2  ...   
Unnamed: 1              2   -3    2    3   -2   -3   -1   -3    2    3  ...   
Unnamed: 2             -1   -2    3    1    3    2   -1   -3   -1    4  ...   
Unnamed: 3             -1   -2    1   -1   -1   -2   -2   -2   -2   -1  ...   
Unnamed: 4              2   -2   -2   -1    0    0   -2   -1   -2   -1  ...   
Unnamed: 5             -1   -3    0   -2    1    0   -1   -4   -3   -1  ...   
Unnamed: 6             -1   -3   -1   -1   -1   -4   -3   -4   -2   -2  ...   
Unnamed: 7             -1   -4   -1   -1    1    1   -3   -4   -2    0  ...   
Unnamed: 8              4   -4    2   -4   -2   -2   -3   -4   -4   -2  ...   
Unnamed: 9             -2   -2   -1   -2    1   -4   -3    2   -3    3  ...   
Unnamed: 10             3   -1    2   -3    3   -1   -1   -4   -2   -2  ...   
Unnamed: 11            -2   -3    3    0    2   -2   -1   -3   -3   -2  ...   
Unnamed: 12            -3   -2   -3   -3   -3   -3    1   -3   -3   -3  ...   
Unnamed: 13            -1   -4   -2    1    1   -2   -3   -3   -2    0  ...   
Unnamed: 14            -1   -4   -3   -3   -3   -2   -4   -4   -3   -2  ...   
Unnamed: 15             1    0    3    1    4    4    3    2    2    1  ...   
Unnamed: 16             1   -1   -1   -2    2    1    1   -3   -3   -1  ...   
Unnamed: 17            -1   -1    3    2    2   -2    1   -2    1    3  ...   
Unnamed: 18            -1   -2    0   -2   -1    1   -1   -4   -2    1  ...   
Unnamed: 19             1   -2    1    2    2   -1   -1   -4   -2    1  ...   
Unnamed: 20             0   -1    0    0    0   -2    2   -1   -1   -1  ...   
Unnamed: 21            -1   -2    0    0   -1    0   -2   -2   -2   -1  ...   
Unnamed: 22             1   -2    2    1    2   -1    1   -2   -1   -2  ...   
Unnamed: 23            -4   -1   -1   -1    1    2   -1    2    1    0  ...   
Unnamed: 24            -1   -1    2    1    2    2    2   -1    1    3  ...   
Unnamed: 25             2   -4   -1   -1   -2   -2   -2   -3   -2   -2  ...   
Unnamed: 26             1   -1    0   -1    1    1    2   -3    0   -1  ...   
Unnamed: 27            -2   -4    3    1    2    2   -1   -4   -2    1  ...   
Unnamed: 28             4   -1    4    1    3    2    2    1    2    2  ...   
Unnamed: 29            -3   -4   -1   -3    3    2   -3   -4   -4    2  ...   

                      100  101  102  103  104  105  106  107  108  109  
Pleasantness Ratings   -1   -1   -1   -1   -1   -4   -2   -1   -1   -3  
Unnamed: 1              3    3    1    2    2   -4    2    2   -2   -2  
Unnamed: 2              1    1    1    1   -2   -4   -2   -3   -2   -4  
Unnamed: 3              2    1   -2    1   -2   -3   -2   -2   -2   -2  
Unnamed: 4              0    2   -3   -1    0   -4   -2    0    0   -3  
Unnamed: 5              1    0   -1   -1    0   -4   -3   -3   -3   -3  
Unnamed: 6              1   -2   -2    1   -2   -4   -1   -2   -2   -4  
Unnamed: 7              1    2   -2    3    0   -4   -1   -3   -1   -4  
Unnamed: 8              4    1   -3    4   -3   -4    1    3   -4   -4  
Unnamed: 9              2    2    2    2   -1   -4   -2    1   -2   -3  
Unnamed: 10             1    1   -1   -1   -2   -4   -1   -2   -3   -4  
Unnamed: 11            -2    2    2   -1    2   -4   -1   -1   -3   -3  
Unnamed: 12             0   -1   -2    0   -2   -3   -3   -3   -3   -2  
Unnamed: 13             2    1    1   -1   -1   -4   -2   -1   -1   -4  
Unnamed: 14            -2   -2   -3    0   -2   -4   -3   -1   -3   -4  
Unnamed: 15             3    4    0    4    2    4    3    3    2   -1  
Unnamed: 16             2    2    1    2   -3   -2    1   -1   -2   -3  
Unnamed: 17            -1    1   -1    2   -2   -2    1   -1    4    3  
Unnamed: 18             2    1    1    1   -2   -3    0   -2   -1   -2  
Unnamed: 19             1    1    1    2    2   -3    0    1    1   -4  
Unnamed: 20             2    2    1    

In [35]:
pg.cronbach_alpha(data=df1_t)

(0.9738331711990418, array([0.959, 0.986]))

## The cronbach alpha for pleasantness is 0.973
## Since Cronbach alpha is greater than 0.9 the internal consistency is excellent

In [36]:
df2_t = df2.T
df2_t

0    1    2    3    4    5    6    7    8    9    ...  100  \
Energy Ratings    1    1    1    1    1    1    1    2    2    2  ...    1   
Unnamed: 1        2    3   -2   -2    2    3    2    4    2    3  ...   -3   
Unnamed: 2        4    3   -1    2   -1    2    3    2    3   -2  ...   -2   
Unnamed: 3        1   -1   -1   -1   -1    1    2    1    2    1  ...   -1   
Unnamed: 4        2    1    1    1   -1    3    1    1    1    0  ...    2   
Unnamed: 5        1    2   -2   -2   -3   -1   -2    1    1    2  ...   -3   
Unnamed: 6       -1    4   -1   -1    2   -3   -1    1    3    2  ...   -1   
Unnamed: 7        2   -1   -3    2   -1   -2    2   -2    1    0  ...   -1   
Unnamed: 8        1    3   -1    3   -2   -1    3    4    4   -1  ...   -3   
Unnamed: 9       -2    2    1    2   -2   -2    2   -2    2    1  ...   -2   
Unnamed: 10       3    3   -2    3   -3    3    2    3    3    3  ...   -2   
Unnamed: 11       2    2    3    1    1   -1    4    3    2    1  ...    1   
Unnamed: 12      -3   -2    2    2   -3   -3    1    1   -3   -3  ...   -1   
Unnamed: 13       3    3    3    2    2    2    3    2    3    1  ...    2   
Unnamed: 14       1    4   -1   -1   -1    1    3    3    2   -1  ...   -1   
Unnamed: 15      -2    4    3    2   -3   -3    4    3    4    2  ...   -3   
Unnamed: 16       3    3    1   -1    2    1    1   -1    3   -1  ...    2   
Unnamed: 17       3    2    2    1    2    1    3    0    2    2  ...    1   
Unnamed: 18      -1    2    0   -2   -1    1   -1   -3   -2    1  ...    2   
Unnamed: 19       2   -2   -1    2   -2    1    1    2    1   -1  ...   -1   
Unnamed: 20       2    1   -1    1   -1    0    0    1    1   -2  ...   -2   
Unnamed: 21       2    2    1    0    0    0    3    1    1    2  ...   -1   
Unnamed: 22       2    3   -1    2    1    1    2    1    2    2  ...    1   
Unnamed: 23      -2    3    2    2    1    2    2    3    3    2  ...   -2   
Unnamed: 24       2    1   -1    1    2   -2    2    3    2    1  ...   -2   
Unnamed: 25       3    3    3    3    1    2    4    3    2    3  ...   -3   
Unnamed: 26       1   -2    0    1    0   -2    2   -1    3    1  ...   -1   
Unnamed: 27       1    4   -2    1   -1   -2    2    3    3   -2  ...   -1   
Unnamed: 28       3    1   -1    2    1    2    3    3    3    2  ...   -1   
Unnamed: 29       4    2   -2    3   -3   -1    3    2    4   -1  ...    0   

                101  102  103  104  105  106  107  108  109  
Energy Ratings   -1    1    1    1    2    2    1    2    2  
Unnamed: 1       -3   -2    2    2    4    1   -3    2    4  
Unnamed: 2        1    2    2    4    4    4    3    4    4  
Unnamed: 3        1    1    1   -1   -2    1    2    1    1  
Unnamed: 4        0    1    1    0   -2    1    0    0    0  
Unnamed: 5       -2    2    2   -1    4   -1   -2   -1    2  
Unnamed: 6        1    2    1    1    4   -1    1    1    4  
Unnamed: 7       -1    2   -1    2    3    4    2    2    3  
Unnamed: 8       -3    3   -4    3    4   -3    1    4    4  
Unnamed: 9       -1    2    1    1    3   -2   -1    2    2  
Unnamed: 10       1    4    2    3    2    2    3    3    3  
Unnamed: 11      -2    2   -2    2    3    3    2    4    3  
Unnamed: 12       1   -2    0    1    2   -3    3    3    2  
Unnamed: 13       2    1    0    2    3    3    2    2    3  
Unnamed: 14       0    2    2    2    4    2    3    3    4  
Unnamed: 15       2    3    2    3    4    1    2    4    3  
Unnamed: 16       0    1    2    3   -1    2    1    3   -2  
Unnamed: 17       0    2    2    1    2    2    1    4    4  
Unnamed: 18       1   -1    1   -1   -3    0   -1   -1   -2  
Unnamed: 19       1    1   -1    1    0   -1    2    2   -1  
Unnamed: 20      -1   -1   -1    0    2    1    1    2    2  
Unnamed: 21      -1    2   -2    1    2    1    1    2    3  
Unnamed: 22      -1    2    2    2    2    1    2    3    3  
Unnamed: 23      -1    2    1    1    2    1    1    2    3  
Unnamed: 24      -1    2    2    2    3    2   

In [37]:
pg.cronbach_alpha(data=df2_t)

(0.9064057081279493, array([0.852, 0.948]))

# The cronbach alpha for energy is 0.906
## Therfore the cronbach alpha is greater than 0.9 which means that internal consistency is excellent

In [38]:
def remove_outlier_IQR(df):
    Q1=df.quantile(0.25)
    Q3=df.quantile(0.75)
    IQR=Q3-Q1
    df_final=df[~((df<(Q1-1.5*IQR)) | (df>(Q3+1.5*IQR)))]
    return df_final

In [39]:
df1t_new = remove_outlier_IQR(df1_t)
pg.cronbach_alpha(data=df1t_new)

(0.9702537456558883, array([0.953, 0.984]))

# Cronbach alpha decreased to 0.970 after removal of outliers

In [40]:
df2t_new = remove_outlier_IQR(df2_t)
pg.cronbach_alpha(data=df2t_new)

(0.9011590977241086, array([0.844, 0.945]))

# Cronbach alpha decreased to 0.901 after removal of outliers